modules/bridge/src/main/ts/ephox/bridge/components/view/ViewButton.ts

```diff
@@ -10,6 +10,7 @@ interface ViewButtonApi {
 interface ViewToggleButtonApi extends ViewButtonApi {
   isActive: () => boolean;
   setActive: (state: boolean) => void;
+  focus: () => void;
```

Comment:
I really regret that views were implemented differently to dialogs, so methods like this have to go here instead of on `View`, but it's too late to change it now 😞

Reply:
Yes it was supposed to be more like the sidebar just a div where we could dump things and the UI would be very minimal but since that things have been added. But I guess since we are going all react now we could just rewrite the UI in react if we want to add more things to it. It's not a complex UI in the advcode plugin, but then again I thought a comment react UI wouldn't as complex as it turned out to be. 😄

Reply:
React won't solve API issues, though. It might be a good opportunity to redo the View API in TinyMCE 8; I doubt anyone is using it other than us.

Reply:
It would since this api would not be needed. We can remove the whole buttons api for views on tinymce v8. Since we only use it for the buttons in the advcode plugin the revisionhistory already does the whole UI in the view api with react.